src/data/prod/basic.lean

```diff
@@ -260,3 +260,55 @@ lemma involutive.prod_map {f : α → α} {g : β → β} :
 left_inverse.prod_map
 
 end function
+
+namespace prod
+open function
+
+@[simp] lemma map_injective [nonempty α] [nonempty β] {f : α → γ} {g : β → δ} :
+  injective (map f g) ↔ injective f ∧ injective g :=
+⟨λ h, ⟨λ a₁ a₂ ha, begin
+  inhabit β,
+  injection @h (a₁, default) (a₂, default) (congr_arg (λ c : γ, prod.mk c (g default)) ha : _),
+end, λ b₁ b₂ hb, begin
+  inhabit α,
+  injection @h (default, b₁) (default, b₂) (congr_arg (prod.mk (f default)) hb : _),
+end⟩, λ h, h.1.prod_map h.2⟩
+
+@[simp] lemma map_surjective [nonempty γ] [nonempty δ] {f : α → γ} {g : β → δ} :
+  surjective (map f g) ↔ surjective f ∧ surjective g :=
+⟨λ h, ⟨λ c, begin
+  inhabit δ,
+  obtain ⟨⟨a, b⟩, h⟩ := h (c, default),
+  exact ⟨a, congr_arg prod.fst h⟩,
+end, λ d, begin
+  inhabit γ,
+  obtain ⟨⟨a, b⟩, h⟩ := h (default, d),
+  exact ⟨b, congr_arg prod.snd h⟩,
+end⟩, λ h, h.1.prod_map h.2⟩
+
+@[simp] lemma map_bijective [nonempty α] [nonempty β] [nonempty γ] [nonempty δ]
```

Comment:
I'm a bit confused myself, but can't you remove one or two `nonempty` hypotheses while keeping the statement true.

Reply:
I removed the last two. I don't think they can be all be removed, consider what happens if `f = @id empty`

Reply:
Yeah, unfortunately this is a situation where 4 different statements up to the hypotheses are true.